File: Napakalaki_Java/src/napakalaki/NumericBadConsequence.java

package napakalaki;

import java.util.ArrayList;


public class NumericBadConsequence extends BadConsequence {
    
    protected int nVisibleTreasures;
    protected int nHiddenTreasures;
    
    public NumericBadConsequence(String text, int levels, int nVisible, int nHidden) {
        super(text, levels);
        this.nVisibleTreasures = nVisible;
        this.nHiddenTreasures = nHidden;
    }
    
    
    @Override
    public BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> visibles, ArrayList<Treasure> hidden) {
        BadConsequence bcAjustado;
        int nVisible = (visibles.size() > nVisibleTreasures ? nVisibleTreasures : visibles.size());
        int nHidden = (hidden.size() > nHiddenTreasures ? nHiddenTreasures : hidden.size());
        bcAjustado = new NumericBadConsequence(text, 0, nVisible, nHidden);

        return bcAjustado;
    }
    
    @Override
    public boolean isEmpty(){
        if(nVisibleTreasures == 0 && nHiddenTreasures == 0)
            return true;
        else
            return false;
    }
    
    public int getNVisibleTreasures() {
        return nVisibleTreasures;
    }

    public int getNHiddenTreasures() {
        return nHiddenTreasures;
    }

    @Override
    public String toString(){
        String description = "--- Mal Rollo Numérico --- \n" + "Texto: " + text + "\n";
        
        if (levels != 0)
            description += "Pérdida de niveles: " + levels + "\n";
        
        if(nVisibleTreasures != 0)
            description += "Pérdida de tesoros visibles: " + nVisibleTreasures + "\n";
        if(nHiddenTreasures != 0)
            description += "Pérdida de tesoros ocultos: " + nHiddenTreasures + "\n";
    
        return description;
    }
}

File: Napakalaki_Ruby/pkg/Napakalaki_Ruby-0.0.1/lib/numeric_bad_consequence.rb
#encoding: utf-8

=begin
  Marco Manuel Fernández Pranno, 2015 (Granada)
  Para la asignatura PDOO (Universidad de Granada)
  Licencia GPLv3 a partir del 15/02/16, hasta entonces todos los derechos reservados.
=end 

class NumericBadConsequence < BadConsequence
  
  def initialize(text, levels, nVisible,nHidden)
    super(text,levels)
    @nVisibleTreasures = nVisible
    @nHiddenTreasures = nHidden
  end
  
  def isEmpty()
    if(@nVisibleTreasures == 0 && @nHiddenTreasures == 0)
      return true
    end
      return false
  end
  
  def adjustToFitTreasureLists(visibles, hidden)
      nVisible = [visibles.size,@nVisibleTreasures].min
      nHidden = [hidden.size,@nHiddenTreasures].min
      return NumericBadConsequence.new(@text,0,nVisible,nHidden)
  end
  
    def to_s
      "--- Bad Consequence Numerico --- \n
      Texto: #{@text} \n Niveles: #{@levels} \n
      nVisibles: #{@nVisibleTreasures} \n
      nHidden: #{@nHiddenTreasures} \n"
    end
    
    attr_reader :nVisibleTreasures, :nHiddenTreasures
    public_class_method :new
end
